### PR TITLE
Update embedding and the code generally

### PR DESCRIPTION
**Pull request by JoelPasvolsky**
Updates this example on a few points:
* Replaces pre-calculated embedding with automatic embedding based on improved default chain strength for D-Wave 2000Q QPUs and shorter chains for Advantage QPUs
* Drop Python 2 support
* Fix setup.py

### CODE REVIEW

**Comment by arcondello** [comment]
I would actually remove the setup and the package info altogether. Not sure why this needs to be a package.

**Comment by randomir** [comment]
Yes, we should absolutely reformat this into a non-package form.

**Comment by JoelPasvolsky** [comment]
I had actually removed the setup & package info but then I worried that maybe it was relevant to its use in the demo. I'll remove. Please assign reviewers.

**Comment by arcondello** [comment]
What about even another step forward, combining `demo.py` and `interfaces.py`?

**Comment by JoelPasvolsky** [comment]
Sure, I can do that. I had actually considered a more extensive update but then this example would become rather similar to https://github.com/dwave-examples/factoring-notebook, which made me wonder whether we wanted to keep both around post-IDE support.

**Comment by arcondello** [comment]
Don't need the `MANIFEST.in` file anymore

**Review by arcondello — APPROVED** [review]
Could probably even take another simplification pass, but it's already *much* better than it was before so LGTM

**Comment by JoelPasvolsky** [comment]
Yes, that is what I meant in https://github.com/dwave-examples/factoring/pull/31#issuecomment-716540643. Any thoughts on post-IDE?
Removed manifest.

**Review by randomir — APPROVED** [review]
This is a great improvement over the old version.

I think we can also remove `__init__.py` from the root dir.

And what do you say we get rid of the whole jsonschema stuff as well? I'm really not sure what value that adds here.

**Comment by randomir** [comment]
Re: post-IDE, we still have some ways to go before we enable jupyter notebook support in the IDE. So, at least until then, I think it makes sense to keep both demos around.

**Comment by JoelPasvolsky** [comment]
Removed json schema.

File: demo.py [hunk 1]
```diff
@@ -42,6 +43,109 @@ def sanitised_input(description, variable, range_):
...
+
+    output = {
+        "Results": [],
+        #    {
+        #        "a": Number,
+        #        "b": Number,
+        #        "valid": Boolean,
+        #        "numOfOccurrences": Number,
+        #        "percentageOfOccurrences": Number
+        #    }
```
Comment (randomir) [inline demo.py]: If we're keeping these comments, then we should update them as well.

File: tests/test_interfaces.py [hunk 2]
```diff
@@ -14,19 +14,11 @@
 
 import unittest
 from random import randint
-import jsonschema
-
-from factoring.interfaces import factor
-from factoring.json_schema import json_schema
 
+from demo import factor
```
Comment (randomir) [inline tests/test_interfaces.py:19]: I'm confused about the purpose of `test_factor_validity` below? If it's always skipped why do we need it? Or if we need it, then let's not always skip it.

Comment (JoelPasvolsky) [reply tests/test_interfaces.py:19]: I unskipped it. There is a small probability of failure, so I give it three chances for each factor.

File: tests/test_interfaces.py [hunk 3]
```diff
@@ -14,25 +14,18 @@
...
 
-    @unittest.skip("Not really a unittest")
     def test_factor_validity(self):
         for P in [12, 21, 49]: # {a*b for a in range(2**3) for b in range(2**3)}:
-            output = factor(P)
-            self.assertTrue(output['results'][0]['valid'])
+            retries = 3
+            while retries > 0 and not factor(P)['Results'][0]['Valid']:
+                retries -= 1
+            self.assertGreater(retries, 0)
```
Comment (randomir) [inline tests/test_interfaces.py]: How do you feel about implementing this with the `@dwave.cloud.utils.retried(3)` decorator?

Comment (JoelPasvolsky) [reply tests/test_interfaces.py]: I feel excited and honored, sir.

Comment (randomir) [reply tests/test_interfaces.py]: I couldn't possibly ask for anything more.

Comment (JoelPasvolsky) [reply tests/test_interfaces.py]: In that case, maybe you'll be good enough to merge this PR.

**Pull request merged**